### PR TITLE
[Stack Overflow] add stackoverflow to plugin marketplace 

### PR DESCRIPTION
**Pull request by emmaindal**
## Hey, I just made a Pull Request!

This PR adds stack overflow plugin to plugin marketplace. 
Screenshot 2022-06-14 at 13 17 28

It also adds an optional icon prop to the homepage component so that its possible to pass in the StackOverflow icon.
Screenshot 2022-06-14 at 15 55 29

With the change of wrapping the entire list item in the Link component, we also fixed https://github.com/backstage/backstage/issues/11674 in this PR.

Adds a storybook example for how the Stackoverflow icon can be used for the search component too.
Screenshot 2022-06-14 at 16 34 17

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: microsite/data/plugins/stack-overflow.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: Stack Overflow
+author: Spotify
+authorUrl: https://github.com/spotify
+category: Discovery
+description: Provides Stack Overflow specific functionality that can be used in different ways (e.g. for homepage and search) to compose your Backstage App.
+documentation: https://github.com/backstage/backstage/blob/master/plugins/stack-overflow
+iconUrl: img/stack-overflow-logo.svg
```
Comment (Rugvip) [inline microsite/data/plugins/stack-overflow.yaml:8]: Feeling that https://stackoverflow.com/legal/trademark-guidance doesn't give us a clear enough permission to do this tbh. Perhaps either check with them if you haven't already, or if we could link directly to an external resource?

File: plugins/stack-overflow/src/icons/StackOverflowIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      height="25"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <path d="M84.021 75.825v31H11v-31H0v42h95v-42H84.021z" fill="#BCBBBB" />
+      <path
+        d="M21.057 96.825H74v-10H21.057v10zm1.058-23.915l51.428 10.794 2.117-10.265L24.23 62.645 22.115 72.91zm6.773-24.656l47.618 22.222 4.445-9.524L33.33 38.73l-4.444 9.524zm13.227-23.386l40.423 33.65 6.773-8.042-40.53-33.65-6.666 8.042zM68.147 0L59.68 6.243l31.323 42.222 8.465-6.243L68.147 0z"
+        fill="#F48024"
+      />
+    </svg>
```
Comment (Rugvip) [inline plugins/stack-overflow/src/icons/StackOverflowIcon.tsx:36]: Same issue over here

**Review by Rugvip — COMMENTED** [review]
Legal stuff 😞

File: microsite/data/plugins/stack-overflow.yaml (hunk 1 above)
Comment (emmaindal) [reply microsite/data/plugins/stack-overflow.yaml:8]: Hey, I've checked with them (mail convo) and the logo was sent over by them and its all approved. Do we need some written approval for it too?

File: plugins/stack-overflow/src/icons/StackOverflowIcon.tsx (hunk 2 above)
Comment (emmaindal) [reply plugins/stack-overflow/src/icons/StackOverflowIcon.tsx:36]: See my comment above!

File: microsite/data/plugins/stack-overflow.yaml (hunk 1 above)
Comment (Rugvip) [reply microsite/data/plugins/stack-overflow.yaml:8]: Sounds good enough! 👍

**Pull request merged**